### PR TITLE
Update plist to 1.2, sort keys on save

### PR DESCRIPTION
**Pull request by cmyr**
I think I've added the sorting code in all the correct places?

- the font's lib
- the layer's lib
- the glyph's lib

Is there anything I'm missing? This is only needed in places where we're
dealing with an arbitrary dictionary, in other places serde will
serialize the keys in the correct order.

progress on #145

### CODE REVIEW

**Comment by chrissimpkins** [comment]
It looks fontinfo.plist was already alphabetized, similar to the approach in ufonormalizer

**Comment by chrissimpkins** [comment]
These sources have norad v0.4 run over them if it is helpful:

https://github.com/chrissimpkins/ufo-format-diff/pull/1/files

**Comment by cmyr** [comment]
> It looks like the following are already in alphabetic order:
> 
> * fontinfo.plist
> * glyphs*/contents.plist appears to already be alphabetized with upper case sorted before lower case
> * groups.plist
> * kerning.plist
> * metainfo.plist

This sounds right, all of those files are deserialized into types that are sorted, and that sorting is maintained when we serialize again. The only thing this was not true for is things that norad represents as dictionaries, and these are only things like the various `lib` objects that are freeform.

**Comment by chrissimpkins** [comment]
*Adding this here for documentation of how the new lib key ordering will take place*

[The plist sort_keys approach](https://docs.rs/plist/1.2.0/plist/dictionary/struct.Dictionary.html#method.sort_keys) supports lexicographic ordering by Unicode code point value as defined in the [`str` type docs](https://doc.rust-lang.org/nightly/std/primitive.str.html#impl-Ord):

> Strings are ordered lexicographically by their byte values. This orders Unicode code points based on their positions in the code charts. This is not necessarily the same as “alphabetical” order, which varies by language and locale. Sorting strings according to culturally-accepted standards requires locale-specific data that is outside the scope of the str type.

File: src/glyph/serialize.rs [hunk 1]
```diff
@@ -79,13 +79,15 @@ impl Glyph {
...
         // cloning and write out the original.
+        let mut lib = self.lib.clone();
         let object_libs = self.dump_object_libs();
         if !object_libs.is_empty() {
-            let mut new_lib = self.lib.clone();
-            new_lib.insert(PUBLIC_OBJECT_LIBS_KEY.into(), plist::Value::Dictionary(object_libs));
-            write_lib_section(&new_lib, &mut writer)?;
-        } else if !self.lib.is_empty() {
-            write_lib_section(&self.lib, &mut writer)?;
+            lib.insert(PUBLIC_OBJECT_LIBS_KEY.into(), object_libs.into());
```
Comment (chrissimpkins) [inline src/glyph/serialize.rs:85]: The changes as of https://github.com/linebender/norad/pull/147/commits/1ee9066f31ff21a5e44d48b903ad092dbbdb41b8 seem to be writing empty `public.objectLibs` dictionary fields in the lib.plist files for some reason

https://github.com/chrissimpkins/ufo-format-diff/commit/6633c6bc2a9f8a0f8527c4383510068d3ed6a10c

File: src/font.rs [hunk 2]
```diff
@@ -251,17 +251,16 @@ impl Font {
...
-            let mut new_lib = self.lib.clone();
-            new_lib.insert(PUBLIC_OBJECT_LIBS_KEY.into(), plist::Value::Dictionary(object_libs));
-            plist::Value::Dictionary(new_lib).to_file_xml(path.join(LIB_FILE))?;
-        } else if !self.lib.is_empty() {
-            plist::Value::Dictionary(self.lib.clone()).to_file_xml(path.join(LIB_FILE))?;
+        // font.lib in-memory. Instead we clone the lib, add the object libs, and
+        // write out that.
+
+        let mut lib = self.lib.clone();
+        if let Some(object_libs) = self.font_info.as_ref().map(FontInfo::dump_object_libs) {
```
Comment (chrissimpkins) [inline src/font.rs:258]: It looks like `FontInfo::dump_object_libs` may be able to return an empty plist Dictionary type and the `if !object_libs.is_empty()` gate was removed?

Comment (cmyr) [reply src/font.rs:258]: thanks!

**Pull request merged**